hbase-server/src/main/java/org/apache/hadoop/hbase/wal/RecoveredEditsOutputSink.java

```diff
@@ -89,20 +89,68 @@ private RecoveredEditsWriter getRecoveredEditsWriter(TableName tableName, byte[]
 
   @Override
   public List<Path> close() throws IOException {
-    boolean isSuccessful = true;
+    boolean isSuccessful;
     try {
       isSuccessful = finishWriterThreads();
-    } finally {
-      isSuccessful &= closeWriters();
+    } catch (InterruptedIOException e) {
```

Comment:
Should this be `IOException` rather than `InterrptedIOException`? Since that is what `finishWriterThreads` may throw.

Reply:
Yes, that should be okay since in the race it would rethrow InterruptedIOException, which would still be caught.